### PR TITLE
Documentation updates

### LINKED ISSUE #1596: How to add users when deploying from script?
**Is your feature request related to a problem? Please describe.**
I'm looking for advice on how to add users after the initial deployment when using the [deploy-from-script](https://github.com/trailofbits/algo/blob/master/docs/deploy-from-script-or-cloud-init-to-localhost.md) method.

**Describe the solution you'd like**
Extra docs in the deploy from script RE how to manage users when using that method. Ideally this method should also not be interactive and be something I can easily script as part of a terraform run or other provisioning.

**Describe alternatives you've considered**
The main method of running the algo script to add users doesnt seem to apply when installing by script. My `/opt/algo` directory doesn't contain my user list; the only reference of the users I'm adding is the resulting `configs/` output. 

Connecting to the server, setting my new `USERS` environment variable and re-running the script does generate the new users, but it seems to be replacing the existing config for users that already exists. I want a way to add or remove users, ideally from a script.

I see you have some alpha terraform modules and I have experimented with those, but atm they feel more complicated than this script method.

**Additional context**
I'm looking to use algo as a replacement for OpenVPN in a company AWS VPC environment. I'm using terraform to create an ASG with 1 instance and a launch template with userdata that calls this script:

```
#!/bin/bash
set -eo xtrace

apt-get -y update
apt-get install -y python-pip
pip install awscli

# Associate EIP
instance_id=$(curl http://169.254.169.254/latest/meta-data/instance-id)
aws ec2 associate-address --instance-id "$instance_id" --allocation-id "${aws_eip.algo.id}" --region "${data.aws_region.current.name}"

# Install algo
export METHOD=local
export ENDPOINT=${aws_eip.algo.public_ip}
export USERS="${join(",", local.users)}"
export STORE_PKI=true
curl -s https://raw.githubusercontent.com/trailofbits/algo/master/install.sh | sudo -E bash -x

aws s3 sync --sse aws:kms /opt/algo/configs s3://${aws_s3_bucket.algo_user_configs.bucket}/algo
``` 

Updating my user list would update the user data so that if my instance dies a new one would come up with the up to date list of users, but I still need a solution to add/remove users programmatically to an existing server. If I can find a solution to do it manually, then I could look at having terraform remote exec run this command when my list of users changes so that the active server gets the latest users.

Also as a sidenote, is it possible to keep the same user profiles if the server gets recreated? Or do I have to redistribute profiles to each user when my server is replaced?

### LINKED ISSUE #1600: How to see stats about users?
Good day!

I am not sure if I am in correct section but I have a quistion! 
How can I see information about users on my Ubuntu server? For example if a user is online and how much is the Bandwidth? Is there a web GUI for eample that I can use?

Many thnx,
Bero

### LINKED ISSUE #1606: Missing documentation on how to connect without "client software"
The first paragraph on the main page says "does not require client software on most devices".
What exactly did you mean by this statement?
I assume this does not mean that most devices implement it in hardware, but instead you mean "3rd party client software", but then I scroll down and see that all client instructions refer to Wireguard, which is a 3rd party client to almost all devices I can think of.
How come?

### LINKED ISSUE #1609: Document how to use Algo behind another router using that router's DNS
_broken out of #624…_

**Is your feature request related to a problem? Please describe.**

When I connect to my Algo VPN that is set up within my home network, I can reach other hosts within my home network LAN by IP but not by hostname.

**Describe the solution you'd like**

One of two:

1. A configuration that sets Algo into "home network mode" wherein it uses DNS provided by DHCP.
2. The documentation explains to set `dns_encryption` to `false` and `dns_servers` to `192.168.1.1` or `10.1.1.1` or whatever the DNS server on the LAN is in the appropriate IPv4 and IPv6 fields in the configuration.

**Describe alternatives you've considered**

1. Set the DNS manually on each client. 🙅‍♂ 
2. Set the DNS manually in each client.conf. 🙅‍♂ 
3. Script automatically changing client.conf post-generation. 🙅‍♂ 
4. Modify how Algo generates client.conf in Ansible scripts. 🙅‍♂ 

**Additional context**

See comments on #624.

### LINKED ISSUE #1610: How to apply changes made to the config file after install.
How can I apply changes I've made to the config file (change DNS server, add/remove users, change network settings) without having to reinstall everything?

### LINKED ISSUE #1622: Error using Python 3.8 on MacOS 10.14
Working on MacOS. Installed Python. Installed dependencies. Added users to config. No other edits to config.

Ran ./algo from algo directory. Received the following message:
```
[WARNING]: Could not match supplied host pattern, ignoring: vpn-host


PLAY [localhost] ***************************************************************
ERROR! Unexpected Exception, this is probably a bug: cannot pickle '_io.TextIOWrapper' object
```

### PR DESCRIPTION
**Pull request by TC1977**
## Description

- Fixes the new variable name (`store_cakey` -> `store_pki`) in the deploy from Ansible docs
- Updates "mission statement" paragraph to reflect the primary use of Wireguard (and its client apps)
- Clarifies the FAQ about clients connecting to each other.
- Clarifies the prompts about VPN On Demand apply only to Apple IPsec clients.
- Encourages people to set their config options and review `config.cfg` _before_ deployment.
- Clarifies that servers (mostly) can't be upgraded to take advantage of "new Algo features".
- Adds instructions to the cloud-init docs about updating user list, and adds the correct locations of install log and configs.
- Adds a FAQ about monitoring user activity.
- Documents Python version incompatibilities described in #1622 
- Adds Road warrior instructions to the "deploy to Ubuntu" instructions.
- Reorganizes `config.cfg` and (again) encourages setting options before deployment.

## Motivation and Context

Fixes #1606.
Fixes question asked in #624 (and in the Gitter chat, numerous times)
Attempts to address #1538 (again)
Closes #1596 .
Nice callback to #1291 .
Addresses #1283, #1216, #1555, closes #1600 , and again numerous questions in Gitter chat
Through a time warp, closes #1610 _before_ it was raised.
Closes #1622 .
Closes #1609 .
Documents the "BetweenClients_DROP" option brought up in #1516, #821, #1171 among others.

## How Has This Been Tested?

Only way to test if documentation works is to see if people still have questions.

## Checklist:

- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,7 +21,6 @@ Algo VPN is a set of Ansible scripts that simplify the setup of a personal Wireg
 * Does not support legacy cipher suites or protocols like L2TP, IKEv1, or RSA
 * Does not install Tor, OpenVPN, or other risky servers
 * Does not depend on the security of [TLS](https://tools.ietf.org/html/rfc7457)
-* Does not require client software on most platforms
```
Comment (jackivanov) [inline README.md]: This is still true for IKEv2 and Apple devices, maybe we need to rephrase this just a bit

File: docs/deploy-from-script-or-cloud-init-to-localhost.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 You can use `install.sh` to prepare the environment and deploy AlgoVPN on the local Ubuntu server in one shot using cloud-init, or run the script directly on the server after it's been created. 
 
-The script doesn't configure any parameters in your cloud, so it's on your own to configure related [firewall rules](/docs/firewalls.md), a floating ip address and other resources you may need. The output of the install script (including the p12 and CA passwords) and user config files will be installed into the `/opt/algo` directory.
+The script doesn't configure any parameters in your cloud, so you're on your own to configure related [firewall rules](/docs/firewalls.md), a floating IP address and other resources you may need. The output of the install script (including the p12 and CA passwords) and user config files will be installed into the `/opt/algo` directory. If you need to update users later, change the user list in `/opt/algo/config.cfg` and run `./algo update-users` from that directory.
```
Comment (jackivanov) [inline docs/deploy-from-script-or-cloud-init-to-localhost.md]: It should work in theory, but I've never tested it. Does it work?

**Comment by davidemyers** [comment]
Encouraging people to review `config.cfg` before deployment is a great idea, but it still says `### Advanced users only below this line ###` immediately after the `users` section. We should consider re-ordering `config.cfg` so that only the truly advanced stuff like subnet numbering and cloud provider configuration are at the bottom of the file below this warning. Probably best done as a separate PR.

File: README.md (hunk 1 above)
Comment (powerlemming) [reply README.md]: You should specify it's only for Apple clients and NOT available for Windows

**Comment by TC1977** [comment]
Great feedback. I'm on the road and will try to address this after the weekend.

File: docs/deploy-from-script-or-cloud-init-to-localhost.md (hunk 2 above)
Comment (TC1977) [reply docs/deploy-from-script-or-cloud-init-to-localhost.md]: It works after I fixed it. Have to activate virtualenv, etc.

File: README.md (hunk 1 above)
Comment (TC1977) [reply README.md]: Ok, hope this is better. Also I fixed the spelling of [IPsec](https://blogs.technet.microsoft.com/ianhamer/2006/02/10/an-ipsec-is-an-ipsec-is-an-ipsec/) and WireGuard.

**Comment by TC1977** [comment]
> Encouraging people to review `config.cfg` before deployment is a great idea, but it still says `### Advanced users only below this line ###` immediately after the `users` section. We should consider re-ordering `config.cfg` so that only the truly advanced stuff like subnet numbering and cloud provider configuration are at the bottom of the file below this warning. Probably best done as a separate PR.

@davidemyers Done! See what you think.

**Comment by davidemyers** [comment]
I like it. Just a few suggestions:

I think `unattended_reboot` should be "above the line" so users are encouraged to review it, but it still needs to be disabled by default.

I suggest the Advanced section start with `dnscrypt_servers` and `dns_servers`, which are probably the first things an advanced user will look for.

I also suggest burying network definitions like `strongswan_network`, `wireguard_network`, and `local_service_ip` at the bottom of the file since they'll almost never be changed and this might encourage even advanced users to leave them alone.

Thanks!

**Comment by TC1977** [comment]
> I like it. Just a few suggestions:
> 
> I think `unattended_reboot` should be "above the line" so users are encouraged to review it, but it still needs to be disabled by default.
> 
> I suggest the Advanced section start with `dnscrypt_servers` and `dns_servers`, which are probably the first things an advanced user will look for.
> 
> I also suggest burying network definitions like `strongswan_network`, `wireguard_network`, and `local_service_ip` at the bottom of the file since they'll almost never be changed and this might encourage even advanced users to leave them alone.
> 
> Thanks!

Ok, done! I also changed `False` to `false` for the `keys_clean_all` option, and rewrote the comment.

**Comment by TC1977** [comment]
Any other feedback, or anything else I should do?

**Comment by TC1977** [comment]
Given how many times the `BetweenClients_DROP` flag has come up, including twice in the last week on [the Gitter chat](https://gitter.im/trailofbits/algo), should that be promoted to a FAQ rather than buried in the deploy to Ubuntu instructions?

**Comment by TC1977** [comment]
Trying to keep these changes in sync with the latest commits. Is there a part that I can drop to get the rest of the doc updates in?

**Comment by jackivanov** [comment]
I think we're good

**Pull request merged**